File: C++/0340.cpp
#include <iostream>
using namespace std;


void swapInt(int& num1, int& num2)
{
    int sw = num1;
    num1 = num2;
    num2 = sw;
}


void sort3(int* box)
{
    if (box[0] > box[1])
        swapInt(box[0], box[1]);
    if (box[1] > box[2])
        swapInt(box[1], box[2]);
    if (box[0] > box[1])
        swapInt(box[0], box[1]);
}

int main()
{
    int box1[3], box2[3];

    cin >> box1[0] >> box1[1] >> box1[2] >> box2[0] >> box2[1] >> box2[2];

    sort3(box1);
    sort3(box2);

    if (box1[0] == box2[0] && box1[1] == box2[1] && box1[2] == box2[2])
        cout << "Boxes are equal" << endl;
    else if (box1[0] <= box2[0] && box1[1] <= box2[1] && box1[2] <= box2[2])
        cout << "The first box is smaller than the second one" << endl;
    else if (box1[0] >= box2[0] && box1[1] >= box2[1] && box1[2] >= box2[2])
        cout << "The first box is larger than the second one" << endl;
    else
        cout << "Boxes are incomparable" << endl;


    return 0;
}

File: Python/0340.py
height1, width1, deepth1 = sorted(map(int, input().split()))
height2, width2, deepth2 = sorted(map(int, input().split()))


if height1 == height2 and width1 == width2 and deepth1 == deepth2:
    print('Boxes are equal')
elif height1 <= height2 and width1 <= width2 and deepth1 <= deepth2:
    print('The first box is smaller than the second one')
elif height1 >= height2 and width1 >= width2 and deepth1 >= deepth2:
    print('The first box is larger than the second one')
else:
    print('Boxes are incomparable')
